source/opt/decoration_manager.cpp

```diff
@@ -229,6 +229,35 @@ std::vector<T> DecorationManager::InternalGetDecorationsFor(uint32_t id,
   return decorations;
 }
 
+const ir::Instruction* DecorationManager::GetDecoration(
+    uint32_t id, uint32_t decoration) const {
+  auto decoration_list = id_to_decoration_insts_.find(id);
+  if (decoration_list != id_to_decoration_insts_.end()) {
```

Comment:
I'm slightly tempted to write this in terms of GetDecorationsFor.    But not that tempted.

No change requested.